text/3414-deprecate-spooky-dropck-at-a-distance.md

```diff
@@ -0,0 +1,362 @@
+- Feature Name: `deprecate_spooky_dropck_at_a_distance`
+- Start Date: 2023-02-13
+- RFC PR: [rust-lang/rfcs#3414](https://github.com/rust-lang/rfcs/pull/3414)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Never add outlives requirements for a non-`needs_drop` type. Adjust
+`#[may_dangle]` for the required semantics.
+
+# Motivation
+[motivation]: #motivation
+
+`PhantomData` having dropck behaviour leads to "spooky-dropck-at-a-distance":
+
+This fails to compile:
+
+```rust
+use core::marker::PhantomData;
+
+struct PrintOnDrop<'s>(&'s str);
+impl<'s> Drop for PrintOnDrop<'s> {
+    fn drop(&mut self) {
+        println!("{}", self.0);
+    }
+}
+
+fn to_pd<T>(_: T) -> PhantomData<T> {
+    PhantomData
+}
+
+pub fn foo() {
+    let mut x;
+    {
+        let s = String::from("temporary");
+        let p = PrintOnDrop(&s);
+        x = (to_pd(p), String::new());
+    }
+}
+```
+
+And yet, this compiles:
```

Comment:
```suggestion
And yet, this compiles, because this time the tuple contains a `()` and  not a `String`:
```

Nit: here and elsewhere, it would be helpful to clarify what has changed in the code.